ddl/db_test.go

```diff
@@ -3387,6 +3387,38 @@ func (s *testDBSuite2) TestDDLWithInvalidTableInfo(c *C) {
 	c.Assert(err.Error(), Equals, "[parser:1064]You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 94 near \"then (b / a) end));\" ")
 }
 
+// TestParallelTruncateTableAndAddColumn tests add column when truncate table.
+func (s *testDBSuite5) TestParallelTruncateTableAndAddColumn(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	tk2 := testkit.NewTestKit(c, s.store)
+	tk2.MustExec("use test_db")
+	s.mustExec(c, "use test_db")
+	s.mustExec(c, "create database if not exists test_truncate_table")
+	s.mustExec(c, "drop table if exists t")
+	s.mustExec(c, "create table t(c1 int)")
+	defer s.mustExec(c, "drop table t;")
+	var addColumnErr error
+	var wg sync.WaitGroup
+	hook := &ddl.TestDDLCallback{}
+	hook.OnJobRunBeforeExported = func(job *model.Job) {
+		if job.Type == model.ActionTruncateTable && job.State == model.JobStateNone {
+			wg.Add(1)
+			go func() {
```

Comment:
Is it possible that the truncate table is executed before the add column operation?